Detectors/TRD/base/src/TrackletTransformer.cxx

```diff
@@ -77,50 +76,34 @@ float TrackletTransformer::calculateZ(int padrow)
   return rowPos - rowSize / 2. - middleRowPos;
 }
 
-float TrackletTransformer::calculateDy(int slope, double oldLorentzAngle, double lorentzAngle, double driftVRatio)
+float TrackletTransformer::calculateDy(int slope, double lorentzAngle, double driftVRatio)
 {
   double padWidth = mPadPlane->getWidthIPad();
 
   // temporary dummy value in cm/microsecond
   float vDrift = 1.5464f;
   float driftHeight = mGeo->cdrHght();
 
-  int dYsigned = 0;
+  int slopeSigned = 0;
   if (slope & (1 << (NBITSTRKLSLOPE - 1))) {
-    dYsigned = -((~(slope - 1)) & ((1 << NBITSTRKLSLOPE) - 1));
+    slopeSigned = -((~(slope - 1)) & ((1 << NBITSTRKLSLOPE) - 1));
   } else {
-    dYsigned = slope & ((1 << NBITSTRKLSLOPE) - 1);
+    slopeSigned = slope & ((1 << NBITSTRKLSLOPE) - 1);
   }
+
   // dy = slope * nTimeBins * padWidth * GRANULARITYTRKLSLOPE;
   // nTimeBins should be number of timebins in drift region. 1 timebin is 100 nanosecond
-  double rawDy = dYsigned * ((driftHeight / vDrift) * 10.) * padWidth * GRANULARITYTRKLSLOPE;
-
-  // driftDistance = 3.35
-  float driftDistance = mGeo->cdrHght() + mGeo->camHght();
-
-  float cmSlope = rawDy / driftDistance;
-
-  double calibratedDy = rawDy - (TMath::Tan(lorentzAngle) * driftDistance);
-  calibratedDy += (TMath::Tan(oldLorentzAngle) * driftDistance * driftVRatio) + cmSlope * (driftDistance * (1 - driftVRatio));
-
-  // ALTERNATIVE METHOD
-
-  // double x_anode_hit = driftDistance*driftVRatio/cmSlope;
-  // double y_anode_hit = driftDistance*driftVRatio;
-
-  // double x_Lorentz_drift_hit = TMath::Tan(oldLorentzAngle)*driftDistance*driftVRatio - TMath::Tan(lorentzAngle)*driftDistance;
-  // double y_Lorentz_drift_hit = driftDistance*driftVRatio - driftDistance;
-
-  // double Delta_x_Lorentz_drift_hit = x_anode_hit - x_Lorentz_drift_hit;
-  // double Delta_y_Lorentz_drift_hit = y_anode_hit - y_Lorentz_drift_hit;
-  // double impact_angle_rec = TMath::ATan2(Delta_y_Lorentz_drift_hit,Delta_x_Lorentz_drift_hit);
+  double rawDy = slopeSigned * ((driftHeight / vDrift) * 10.) * padWidth * GRANULARITYTRKLSLOPE;
 
-  // float calibrationShift = TMath::Tan(impact_angle_rec) * driftDistance;
+  // NOTE: check what drift height is used in calibration code to ensure consistency
+  // NOTE: check sign convention of Lorentz angle
+  // NOTE: confirm the direction in which vDrift is measured/determined. Is it in x or in direction of drift?
+  double lorentzCorrection = TMath::Tan(lorentzAngle) * driftHeight;
 
-  // LOG(info) << "ORIGINAL: " << calibratedDy;
-  // LOG(info) << "ALTERNATIVE: " << rawDy + calibrationShift;
+  // assuming angle in Bailhache, fig. 4.17 would be positive in our calibration code
+  double calibratedDy = rawDy - lorentzCorrection;
 
-  return rawDy; // OS: temporary until calibratedDy is checked. Currently it is too far off from rawDy
+  return calibratedDy;
```

Comment:
Please don't yet send the `calibratedDy` but the `rawDy` instead until we actually have proper calibration

Reply:
Or did you already compare calibrated and raw value and check whether they are in sufficient agreement?

Reply:
Well I have a Lorentz angle of 8 degrees hardcoded in there which equates to an offset of about -0.4cm over 3cm. I don't really know whether this would be better or worse for your tracking. I think 8 degrees is a reasonable estimate and hopefully the sign is correct...

Reply:
Maybe you could just run a small simulation and actually compare both values for `dY` which you get?

Reply:
Yeah, I did a quick test and all rawDy values were just reduced by ~0.42cm

Reply:
Given that dy ranges from -0.9 cm to 0.9 cm that is not a negligible change.. So the tracker would not find any matches if dy is shifted by that much. Only when we have our actual FEE configuration and the TrapSimulator produces the compatible output as the FEE only then we can tune the calibration of the tracklets and use the calibrated values. Until then we have to stick with the raw values. At the moment the TrapSimulator still does some correction to the tracklets